### ISSUE TITLE
Extra slashes "\" in Redis cache data

### ISSUE THREAD

**Issue by jainpratham1989** [StackExchange/StackExchange.Redis#1413]
Hi All,

I have a C# program which serializes an object into JSON string and send it to Redis cache and store there. While serialization of object into JSON string it adds extra slashes `"\"` and sends it to Redis and save it into Redis with these extra slashes `"\"`. Please let me know how can I save this JSON message without these extra slashes `"\"` in Redis cache. Below is the code snippet I have written-  
 
``` c#
var setting = new JsonSerializerSettings();
setting.StringEscapeHandling = StringEscapeHandling.EscapeNonAscii;

MyMessage<T> t = new MyMessage<T>(objMessage);

string json = JsonConvert.SerializeObject(t, Formatting.None, setting);

invocationTask = myProxy.Invoke("CreateMessage", MessagePath, json);
```

**Comment by mgravell** [comment]
Redis (server) and SE-Redis (library) just see opaque blobs/clobs, so I doubt anything that is happening is coming from the library or server. However, since you haven't included any code relating to how you access redis, it is hard to be sure.

Also: are you sure there are extra slashes? Where and how exactly are you looking to see this? It is remarkably common to mistake the escape characters that are used *in debug outputs* (but which are not actually there) for extra slashes.

**Comment by bklooste** [comment]
I have the same issue as well and i have non C# consumers. 

`      public static NameValueEntry[] ToRedisEvent(ReadOnlyMemory<byte> utf8Body, string typeName, string metadata)
        {
            return new NameValueEntry[]
            {
                new NameValueEntry(RedisEventTypeName , typeName),
                new NameValueEntry(RedisEventBodyName , utf8Body),
                new NameValueEntry(RedisEventMetaDataName , metadata)
            };
        }`

Now querying with Xrange  from outside C# i get 

{"id":"1586929400544-0","data":{"$t":"BetRegisteredEvent","msg":"{\"totalStake\":100,\"betClientRef\":\"66268539582323173194847552506921\",\"betGroups\" {redacted} \"}","$m":"{\"$c\":\"6cc14827-c7bb-4963-84fd-2464e4c73fc1\"}"}}

However what i expected / desired was 
,"msg":{"totalStake":100,"betClientRef  etc } .
 eg a json object but somehow it became a c# escaped string.  Obviously in C# this will deserialze correctly ( string or object) but  C consumers dont like all the escaping and the content should begin with a { not a " 

bytes passed in to StackExchange.Redis 
123 34 116 111 116

which begins with 123  {  not quote (34)

**Comment by bklooste** [comment]
I think its the web client im doing XRange from  . I can confirm that  RedisValue is RAW

**Comment by mgravell** [comment]
@bklooste it isn't clear from the example, but it *sounds* like you're xrange client is bringing that data back to you as a JSON-esque set of name/value pairs, and that one of your strings *just happens* to contain JSON. That's fine, but that doesn't mean that your client is going to treat it as objects : it is just *a string that happens to contain JSON*. This sounds like an expectation failure rather than a library error, specifically here:

> eg a json object but somehow it became a c# escaped string

(note: this is almost certainly JSON escaped, not C# escaped - they just happen to be the same here)

Basically, whatever client you're using *doesn't know* that you want to materialize it as an object - it is *just a string*

**Comment by bklooste** [comment]
I think your right need to get a client that returns bytes to confirm this.  .  The bytes should not be escaped.

**Comment by mgravell** [comment]
@jainpratham1989 there hasn't been any update here; can you provide any context or repro/example that makes you believe there is a library error here?

**Comment by mgravell** [comment]
SE-Redis doesn't do any escaping, and it you cast the `RedisValue` to `byte[]` you are fetching the raw bytes. But as long as your data doesn't contain formatting characters, you can probably confirm with just `redis-cli`

**Comment by bklooste** [comment]
Confirmed its the reading of the value  , bash also escaped it ..  but redis-cli --raw was correct.

**Comment by mgravell** [comment]
So: nothing to do here?

**Comment by NickCraver** [comment]
Closing out to cleanup
